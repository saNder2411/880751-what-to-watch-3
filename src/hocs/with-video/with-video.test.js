import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withVideo from './with-video.jsx';


const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withVideo(MockComponent);

const videoProps = {
  isPlaying: false,
  poster: ``,
  src: ``,
  isMuted: true,
  isDelay: true,
  width: 280,
  height: 175,
};

it(`withVideo is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped {...videoProps}/>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
