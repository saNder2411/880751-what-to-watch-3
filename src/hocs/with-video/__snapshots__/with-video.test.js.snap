// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideo is rendered correctly for PreviewCard 1`] = `
<video
  className=""
/>
`;

exports[`withVideo is rendered correctly for VideoPlayerScreen 1`] = `
<video
  className="player__video"
/>
`;
